src/common/snippets/src/pass/explicit_transpose_matmul_inputs.cpp

```diff
@@ -2,79 +2,99 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
-#include "snippets/itt.hpp"
-
 #include "snippets/pass/explicit_transpose_matmul_inputs.hpp"
-#include "snippets/pass/transpose_decomposition.hpp"
+
 #include "snippets/op/subgraph.hpp"
+#include "snippets/itt.hpp"
 
-#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/pattern/matcher.hpp"
 #include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "openvino/core/rt_info.hpp"
+
+
+void ov::snippets::pass::ExplicitTransposeMatMulInputs::extract(const ov::Input<ov::Node>& input) {
+    auto parent = input.get_source_output().get_node_shared_ptr();
+    auto transpose = ov::as_type_ptr<ov::op::v1::Transpose>(parent);
+    while (!transpose && !ov::is_type<ov::op::v0::Parameter>(parent)) {
+        // We can set supported order and transposed_<a|b>=false only if ops have scalar shapes to avoid shape mismatching
+        const auto parent_count = parent->inputs().size();
+        bool are_weights_scalar = true;
+        for (size_t j = 1; j < parent_count; ++j) {
+            are_weights_scalar = are_weights_scalar && ov::shape_size(parent->get_input_shape(j)) == 1;
+        }
+        if (!are_weights_scalar)
+            break;
```

Comment:
We have exactly the same code in mha_tokenization.cpp on L308-315. Do we need this logic duplication? If we do, then can we avoid code duplication at least?

Reply:
It's needed to avoid the possible conflicts and bugs in the future. Just in case 🙂 
But you're right. We can avoid code duplication. Done, thanks for the idea!